modules/core/src/main/scala/dk/cwconsult/peregrin/core/impl/MigrationsImpl.scala

```diff
@@ -166,24 +212,27 @@ private[peregrin] class MigrationsImpl(connection: Connection, schema: Schema) {
     * over multiple calls to [[applyChangeLog()]] (as opposed to doing it
     * in a single call -- which is the case this is meant for handling).
     *
-    * We assume that the input list is sorted by identifier.
+    * We assume that the input list is sorted.
     */
   private[this] def removeDuplicates(migrations: Vector[Migration]): Vector[Migration] = {
-    // Check that all adjacent "duplicates" are in fact identical. By the
-    // "sorted" property we know that all duplicates will be adjacent, and
-    // by the transitive property of equality we know that once we're through
+    // Check that all "duplicates" are in fact identical. We group all migrations
+    // with identical identifiers, and iterate through all migrations with identical
+    // identifiers and compare the SQL, so that we know that once we're through
     // this loop *all* migrations with a given identifier must have identical
     // SQL.
-    migrations.zip(migrations.drop(1)).foreach {
-      case (existingMigration, newMigration) =>
-        if (existingMigration.identifier == newMigration.identifier) {
-          if (existingMigration.sql != newMigration.sql) {
-            throwDoesNotMatch(
-              existingSql = existingMigration.sql,
-              newSql = newMigration.sql)
-          }
+    migrations
+      .groupBy(_.identifier)
+      .filter(_._2.size > 1)
+      .foreach { kv =>
+        val differentSqls =
+          kv._2.map(_.sql).distinct
+        if (differentSqls.length > 1) {
```

Comment:
pattern match hellere på differentSqls.headOption, så slipper du for redundant headOption.getOrElse("") nedenfor